### PR TITLE
refactor: Tooltip component

### PR DESCRIPTION
**Pull request by G0L0L**

### CODE REVIEW

File: src/Tooltip/Tooltip.stories.tsx [hunk 1]
```diff
@@ -30,9 +18,22 @@ storiesOf('molecules|Tooltip', module)
       <div>
         <Tooltip
           placement={TooltipPlacement.TOP}
```
Comment (hongaar) [inline src/Tooltip/Tooltip.stories.tsx]: Is there a reason to specify placement explicitly here? Just curious if there's a default prop value

**Comment by Stefanwullems** [comment]
Wouldn't if make more sense if the icon is given by props and the children are used as content instead of the other way around? Otherwise lgtm.

**Comment by hongaar** [comment]
> Wouldn't if make more sense if the icon is given by props and the children are used as content instead of the other way around?

At first I was also confused by this, but actually I like the new approach better. The icon is what will be initially rendered, so it does make sense to me that uses the `children` prop.

Don't feel very strongly about this though...

**Comment by Stefanwullems** [comment]
@hongaar what do you mean by initially rendered?

**Comment by hongaar** [comment]
> @hongaar what do you mean by initially rendered?

As in, the tooltip itself (the contents) will not be rendered/visible initially. Only after a user triggers the tooltip to open by hovering/clicking/etc.

**Comment by Stefanwullems** [comment]
It's the same with the dropdown and the select then. However, I think it's better to just put the main content in children like we do with every other component. We don't instantiate a tooltip to show an icon, we instantiate a tooltip to give some extra information on something.

File: src/Tooltip/Tooltip.stories.tsx (hunk 1 above)
Comment (G0L0L) [reply src/Tooltip/Tooltip.stories.tsx]: It looks logical for me. But, for sure I can make additional prop

**Comment by G0L0L** [comment]
> Wouldn't if make more sense if the icon is given by props and the children are used as content instead of the other way around? Otherwise lgtm.

I've got an inspiration from [material](https://material-ui.com/components/tooltips/) and [mineral](https://mineral-ui.com/components/tooltip).
Actually, I had the same problem, while working on mineral-ui. From programmer perspective, I think it makes more sense to leave it as it is. Since the part you are rendering is trigger and content is just additional info. Try to think about an `<input>` with tooltip (question mark) near it.

File: src/Tooltip/Tooltip.stories.tsx (hunk 1 above)
Comment (hongaar) [reply src/Tooltip/Tooltip.stories.tsx]: Sorry I wasn't clear. What I mean is, is there a difference between

```jsx
<Tooltip placement={TooltipPlacement.TOP} content={...} />
```

and 

```jsx
<Tooltip content={...} />
```

**Comment by hongaar** [comment]
> It's the same with the dropdown and the select then.

Good point. Better to be consistent. You've convinced me 😄

File: src/Tooltip/Tooltip.stories.tsx (hunk 1 above)
Comment (G0L0L) [reply src/Tooltip/Tooltip.stories.tsx]: Oups, I misunderstood the thing. `TOP` is default, I used here just to be explicit.

**Comment by G0L0L** [comment]
> > It's the same with the dropdown and the select then.
> 
> Good point. Better to be consistent. You've convinced me 😄

Imho in this particular case, logic should overcome consistency :)
I can agree, that `Dropdown` might have the same structure, as `Tooltip`. But definitely not `Select`.

**Comment by hongaar** [comment]
@G0L0L I do appreciate that argument too.

(As an aside: Reakit takes an even different approach by rendering the two elements side by side: https://reakit.io/docs/dialog/)

I'm split 🔀 so I'd say go ahead with your proposed changes! ✔️

**Pull request merged**